src/common/transformations/src/transformations/op_conversions/convert_gather_to_compressed.cpp

```diff
@@ -54,95 +55,132 @@ ov::pass::ConvertGatherToGatherCompressed::ConvertGatherToGatherCompressed() {
         std::make_shared<ov::pass::pattern::op::Or>(ov::OutputVector{reshape_m, last_convert_m, mul_m});
     auto gather_m = wrap_type<ov::op::v8::Gather>({dicts_input_m, any_input(), wrap_type<ov::op::v0::Constant>()});
 
+    // fp16/bf16 constant + convert(real16 to f32) + gather case
```

Comment:
If we leave current implementation (without mixed precision support), I don't see any reasons to include the new logic in   `ConvertGatherToGatherCompressed` pass. We have a separate matcher for FP16->FP32 decompression case, and separate transformation callback. So let's create a new pass for that.
To avoid overheads on 2 passes (double model traversal), we can add the both matchers in a single `GraphRewrite`. An example: https://github.com/openvinotoolkit/openvino/blob/b80cecf29dc606c5752cddac4312c383094c805f/src/common/transformations/include/transformations/common_optimizations/gelu_fusion.hpp#L97-L108
Then, this `GraphRewrite` is registered by the plugin in transformation pipeline.

Reply:
Good suggestion. Still in ConvertGatherToGatherCompressed, but two separate matchers with single model traversal. Done, Thanks Vladislav!

Reply:
@dmitry-gorokhov @v-Golubev, The new GraphRewrite pass `CompressedGatherTransformation` has two matcher, `ConvertGatherToGatherCompressed` and `MoveDecompressionAfterGather`. CPU plugin register `CompressedGatherTransformation` including both passes. I does not touch GPU plugin, which still register  `ConvertGatherToGatherCompressed`, if GPU team think there is no risk for the new pass, then they should update to register `CompressedGatherTransformation`. What do you think?

Reply:
@sshlyapn do you have any concerns about the `MoveDecompressionAfterGather`? Please pay special attention to the precisions for which the new transformation works: [f16 and bf16](https://github.com/chenhu-wang/openvino/blob/38573d2babaffe35fdf68bf2ccefcacb558be30e/src/common/transformations/src/transformations/op_conversions/convert_gather_to_compressed.cpp#L154-L156)